shopinvader/models/shopinvader_variant.py

```diff
@@ -293,11 +293,38 @@ def _get_price(
             )
         return res
 
+    @api.model
+    def _get_main_product_read_fields(cls):
+        product_model = cls.env["product.product"]
+        order_by = [x.strip() for x in product_model._order.split(",")]
+        return ["shopinvader_product_id", "backend_id", "lang_id"] + order_by
+
+    @api.model
+    def _get_main_product_sorted_variants(cls, variants):
```

Comment:
I'm not sure I see the value for this hook.
Wouldn't `_pick_main_variant` suffice?